homeassistant/components/zwave_js/__init__.py

```diff
@@ -55,21 +56,31 @@ async def async_on_initialized():
         LOGGER.info("Connection to Zwave JS Server initialized")
         initialized.set()
 
-        # This should be moved to the Node Ready command or something
-        # And then listen to new nodes too.
-        dev_reg = await device_registry.async_get_registry(hass)
+        # run discovery on all ready nodes
         for node in client.driver.controller.nodes.values():
-            register_node_in_dev_reg(entry, dev_reg, client, node)
+            if not node.ready:
+                continue
+            await async_on_node_ready(node)
+        # TODO: register callback for "node_ready" event
 
     async def async_on_disconnect():
         """Handle websocket is disconnected."""
         LOGGER.info("Disconnected from Zwave JS Server")
         # TODO: signal entities to update availability state
 
+    async def async_on_node_ready(node: ZwaveNode):
+        """Handle node ready event."""
+        # register node in device registry
+        dev_reg = await device_registry.async_get_registry(hass)
+        register_node_in_dev_reg(entry, dev_reg, client, node)
+        # run discovery on all node values and create/update entities
+        async for disc_info in async_discover_values(node):
+            LOGGER.info("Discovered value: %s", disc_info)
```

Comment:
```suggestion
            LOGGER.debug("Discovered value: %s", disc_info)
```

Reply:
this will be gone in next commit when I actually dispatch to platforms.
Or do you want it in for debugging purposes ?

Reply:
No, if we don't need it we can remove it.